test/exercises/diffusion/heat_2d_dist_exchanges.chpl

```diff
@@ -0,0 +1,145 @@
+/*
+  A distributed 2D finite-difference heat/diffusion equation solver
+
+  Computation is executed over a 2D distributed array.
+  The array distribution is managed by the `Block` distribution.
+  Tasks are spawned manually with a `coforall` loop and synchronization
+  is done manually using a `barrier`. Halo regions are shared across
+  locales manually via direct assignment between locally owned arrays.
+*/
+
+import BlockDist.Block,
+       Collectives.barrier,
+       Time.stopwatch;
+
+// create a stopwatch to time kernel execution
+var t = new stopwatch();
+config const writeTime = false;
+
+// compile with `-sRunCommDiag=true` to see comm diagnostics
+use CommDiagnostics;
+config param RunCommDiag = false;
+
+// declare configurable constants with default values
+config const nx = 256,      // number of grid points in x
+             ny = 256,      // number of grid points in y
+             nt = 50,       // number of time steps
+             alpha = 0.25,  // diffusion constant
+             solutionStd = 0.222751; // known solution for the default parameters
+
+// define distributed domains and block-distributed array
+const indices = {0..<nx, 0..<ny},
+      indicesInner = indices.expand(-1),
+      Indices = Block.createDomain(indices);
```

Comment:
Trying to figure out for myself the reason for these three domains rather than the same ones in the Block version or others:
* `Indices` is needed because it's used to declare `u` and also to determine what a given locale owns
* `indicesInner` is needed because we're slicing a local domain with it and want the result to be local?  (feels a little subtle / one-use)
* `indices` is used to avoid typing the same domain literal twice in these two declarations?

I think I'd suggest trying to reuse the same declarations as the Block version for consistency / to reduce differences (esp. given your goal of trying to keep the "outermost" aspects of the computation the same and only exposing the SPMD view inside the computation?):

```chapel
const Indices = Block.createDomain({0..<nx, 0..<ny}),
      IndicesInner = Indices[1..<nx-1, 1..<ny-1];
```

and then replace

```chapel
 localIndicesInner = myGlobalIndices[indicesInner];
```

with:

```chapel
 localIndicesInner = IndicesInner.localSubdomain();
```

Reply:
Oops, I meant for the above comment to be on the `buffers` version, though I think it applies to both?  too many windows open...